ras_party/controllers/account_controller.py

```diff
@@ -347,6 +348,74 @@ def resend_verification_email(party_uuid, session):
     return {'message': EMAIL_VERIFICATION_SENT}
 
 
+@transactional
+@with_db_session
+def add_new_survey_for_respondent(payload, tran, session):
+    """
+    Add a survey for an existing respondent
+    :param payload: json containing party_id and enrolment_code
+    :param session: database session
+    """
+    logger.info("Enrolling existing respondent in survey")
+    v = Validator(Exists('party_id'))
+    if not v.validate(payload):
+        raise RasError(v.errors, 400)
+
+    v1 = Validator(Exists('enrolment_code'))
+    if not v1.validate(payload):
+        raise RasError(v1.errors, 400)
+
+    respondent_party_id = payload['party_id']
+    enrolment_code = payload['enrolment_code']
+
+    iac = request_iac(enrolment_code)
+    if not iac.get('active'):
+        raise RasError("Enrolment code is not active.", status_code=400)
+
+    respondent = query_respondent_by_party_uuid(respondent_party_id, session)
+
+    case_context = request_case(enrolment_code)
+    case_id = case_context['id']
+    business_id = case_context['partyId']
+    collection_exercise_id = case_context['caseGroup']['collectionExerciseId']
+    collection_exercise = request_collection_exercise(collection_exercise_id)
+
+    try:
+        survey_id = collection_exercise['surveyId']
+        survey = request_survey(survey_id)
+        survey_name = survey['longName']
+    except KeyError:
+        raise RasError("There is no survey bound for this user with id: ".format(email_address=respondent.id))
```

Comment:
Do we need to check for this `KeyError`? 
Unless there's a particular case that we know of where the collection exercise wont contain a surveyId it probably isn't required

Reply:
Nice to have, so if the response is not what we expected we know exactly what was wrong and it's handled elegantly instead of the application just breaking down.

Reply:
I can see the value but I don't understand why we're doing it for this particular bit of data and nowhere else if we are taking that approach